### PR TITLE
Added functions for calculating sample sizes from either odds ratio or Cohen's w

### PR DESCRIPTION
**Pull request by bearloga**
Includes unit tests :)

### CODE REVIEW

File: R/logs.R [hunk 1]
```diff
@@ -34,21 +34,34 @@ get_logfile <- function(earliest = NULL, latest = NULL){
...
-#'"lang" and "x_analytics".
+#'  "servicetime", "ip_address", "status_code", "reply_size", "request_method",
+#'  "url", "squid_status", "mime_type", "referer", "x_forwarded", "user_agent",
+#'  "lang" and "x_analytics".
 #'
 #'@importFrom urltools url_decode
 #'@export
-read_sampled_log <- function(file){
-  output_file <- tempfile()
-  system(paste("gunzip -c", file, ">", output_file))
```
Comment (Ironholds) [inline R/logs.R]: Awesome changes but they should be described in the commit message

File: R/power.R [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+#'sample_size_odds(odds_ratio = 2, p_control = 0.4)
+#'sample_size_odds(odds_ratio = 2, p_control = 0.4, visualize = TRUE)
+#'
+#'@export
+sample_size_odds <- function(odds_ratio = NULL,
+                             p_control = NULL,
+                             p_treatment = NULL,
+                             power = NULL, conf_level = 0.95,
+                             sample_ratio = 1, visualize = FALSE)
+{ # Checks
```
Comment (Ironholds) [inline R/power.R]: Opening braces never go on newlines

File: R/power.R [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  power_missing <- is.null(power)
+  pC_missing <- is.null(p_control)
+  pT_missing <- is.null(p_treatment)
+  oR_missing <- is.null(odds_ratio)
+  if ( (oR_missing + pC_missing + pT_missing) > 1)
+    stop("Only one of {odds_ratio, p_control, p_treatment} can be missing.")
```
Comment (Ironholds) [inline R/power.R]: If it's long enough it needs a different line it should have braces (I mean I think all control flow statements should have braces but, style guide here)

File: R/power.R [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    } else {
+      warning("All parameters known. Nothing to visualize.")
+    }
+  }
+
+  # Output
+  if (power_missing || length(power) > 1) {
+    names(n) <- sprintf("%.0f%%", power*100)
+    return(n)
+  } else {
```
Comment (Ironholds) [inline R/power.R]: Extraneous statement. You could write it as:

      if(foo){
        names(n) <- bar
      }
      return(n)

File: R/power.R [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+#'@examples
+#'sample_size_effect()
+#'sample_size_effect(0.1)
+#'sample_size_effect(w = 0.1, groups = 3, sig_level = 0.001, power = 0.9)
+#'
+#'@importFrom pwr pwr.chisq.test
+#'@export
+sample_size_effect <- function(w = NULL, groups = 2,
+                               sig_level = 0.05, power = 0.8)
+{ # Checks
```
Comment (Ironholds) [inline R/power.R]: See above

File: R/power.R [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    n <- ceiling(sapply(w, function(ww) {
+      pwr::pwr.chisq.test(w = ww, N = NULL, df = groups - 1,
+                          sig.level = sig_level, power = power)$N
+    }))
+    names(n) <- c("tiny", "small", "medium", "large")
+    return(n)
+  } else {
+    n <- ceiling(pwr::pwr.chisq.test(w = w, N = NULL, df = groups - 1,
+                                     sig.level = sig_level, power = power)$N)
+    return(n)
```
Comment (Ironholds) [inline R/power.R]: Again extraneous else. If(foo){return()} doesn't need an else; if it gets to the else the if-condition was false.

File: R/logs.R (hunk 1 above)
Comment (bearloga) [inline R/logs.R]: Sorry about that

**Comment by Ironholds** [comment]
LGTM

**Pull request merged**